### ISSUE TITLE
Indent relative to inline opening tag

### ISSUE THREAD

**Issue by Rarst** [squizlabs/PHP_CodeSniffer#1143]
I am going over following code and getting indentation error:

```php
<div class="onpage">
	<h3 class="hide-if-no-js"><?php
		printf(
			/* translators: 1: expands to OnPage.org */
			__( 'Indexability check by %1$s', 'wordpress-seo' ),
			'OnPage.org'
		); // line 48
	?></h3>
```

https://github.com/Yoast/wordpress-seo/blob/45f0a41ac5432dbfe799fb1ee26e5a82eeb3148f/admin/views/dashboard-widget.php#L48

```
FILE: ...p-content\plugins\wordpress-seo\admin\views\dashboard-widget.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 48 | ERROR | [x] Line indented incorrectly; expected at least 7 tabs,
    |       |     found 2 (Generic.WhiteSpace.ScopeIndent.Incorrect)
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

It seems to do with it wanting to be relative to opening `<?php` tag, but in this case it's inline rather than blocky...

Is this correct/intended?

**Comment by gsherwood** [comment]
Yes, this is what the Scope Indent sniff wants you to do. It uses PHP open and close tags to define indents. It doesn't parse HTML code to figure out where your HTML tags are and use those for indenting because PHPCS doesn't have a HTML tokenizer built in.

**Comment by Rarst** [comment]
Thank you for information! If it's by design (or at least by technical necessity) then good enough. :)

**Comment by Kolyunya** [comment]
@Rarst do we really want the 
~~~php
<div>
    <?php
    if ($foo) {
        # ...
    } else {
        # ...
    }
    ?>
</div>
~~~
instead of the:
~~~php
<div>
    <?php
        if ($foo) {
            # ...
        } else {
            # ...
        }
    ?>
</div>
~~~
to be valid? Is the first snippet really supposed to be valid or is this a bug?

**Comment by Kolyunya** [comment]
@gsherwood do we really want the 
~~~php
<div>
    <?php
    if ($foo) {
        # ...
    } else {
        # ...
    }
    ?>
</div>
~~~
instead of the:
~~~php
<div>
    <?php
        if ($foo) {
            # ...
        } else {
            # ...
        }
    ?>
</div>
~~~
to be valid? Is the first snippet really supposed to be valid or is this a bug?

**Comment by gsherwood** [comment]
> @gsherwood do we really want... to be valid? Is the first snippet really supposed to be valid or is this a bug?

The fact that you are indenting your code is just because it is inside a template file. You'd never write a standard PHP file like this:
```php
<?php
    if ($foo) {
        # ...
    } else {
        # ...
    }
```
The particular PHPCS sniff checking indents doesn't differentiate between these two types of files, so it is just enforcing what the included standards want. If you don't want that, then you need a different sniff, or for this sniff to have an option to change it's behaviour. That's best submitted as a new feature request though.
